cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -244,18 +242,34 @@ if(builtin_lz4)
     set(LZ4_CFLAGS "-Wno-format-nonliteral")
   elseif( CMAKE_CXX_COMPILER_ID STREQUAL Intel)
     set(LZ4_CFLAGS "-wd188 -wd181 -wd1292 -wd10006 -wd10156 -wd2259 -wd981 -wd128 -wd3179")
+  elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+    set(LZ4_CFLAGS "/Zl")
   endif()
   set(LZ4_LIBRARIES ${CMAKE_BINARY_DIR}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}lz4${CMAKE_STATIC_LIBRARY_SUFFIX})
-  ExternalProject_Add(
-    LZ4
-    URL ${lcgpackages}/lz4-${lz4_version}.tar.gz
-    URL_MD5 c9610c5ce97eb431dddddf0073d919b9
-    INSTALL_DIR ${CMAKE_BINARY_DIR}
-    CONFIGURE_COMMAND  /bin/sh -c "PREFIX=<INSTALL_DIR> CMAKE_PARAMS='-DCMAKE_C_COMPILER=\\\"${CMAKE_C_COMPILER}\\\" -DCMAKE_C_FLAGS=\\\"${CMAKE_C_FLAGS}\\\" -DCMAKE_OSX_SYSROOT=\\\"${CMAKE_OSX_SYSROOT}\\\"' make cmake"
-    BUILD_COMMAND /bin/sh -c "PREFIX=<INSTALL_DIR> MOREFLAGS=-fPIC make"
-    INSTALL_COMMAND /bin/sh -c "PREFIX=<INSTALL_DIR> make install"
-    LOG_DOWNLOAD 1 LOG_CONFIGURE 1 LOG_BUILD 1 LOG_INSTALL 1 BUILD_IN_SOURCE 1
-    BUILD_BYPRODUCTS ${LZ4_LIBRARIES})
+  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+    file(TO_NATIVE_PATH "${CMAKE_BINARY_DIR}/include" NATIVE_INCLUDEDIR)
+    ExternalProject_Add(
+      LZ4
+      URL http://lcgpackages.web.cern.ch/lcgpackages/tarFiles/sources/lz4-${lz4_version}.tar.gz
```

Comment:
Maybe it's worth to avoid duplication of URL and MD5 hash. How about
```cmake
set(LZ4_URL ${lcgpackages}/lz4-${lz4_version}.tar.gz)
set(LZ4_MD5 c9610c5ce97eb431dddddf0073d919b9)
```
and then use the variables in both branches? That way we only have one place to update if something changes.

Reply:
Good point, thanks! Will do it